packages/webapp/src/elections/components/registration-election-components/participation-card.tsx

```diff
@@ -161,6 +196,41 @@ const ConfirmParticipationModal = ({ isOpen, close }: ModalProps) => {
         setIsLoading(false);
     };
 
+    const submitParticipationConfirmation = () => {
+        if (!encryptionPassword.privateKey) {
+            setStep(ParticipationStep.ConfirmPassword);
+        } else {
+            onSubmit();
+        }
+    };
+
+    const submitPasswordConfirmation = async (
+        setEncryptionPasswordTrx?: any
+    ) => {
+        await onSubmit(setEncryptionPasswordTrx);
+    };
+
+    const renderStep = () => {
+        switch (step) {
+            case ParticipationStep.ConfirmParticipation:
+                return (
+                    <ConfirmParticipationStep
+                        onSubmit={submitParticipationConfirmation}
+                        isLoading={isLoading}
+                        onCancel={close}
+                    />
+                );
+            case ParticipationStep.ConfirmPassword:
+                return (
+                    <ConfirmPasswordStep
+                        onSubmit={submitPasswordConfirmation}
+                        isLoading={isLoading}
+                        onCancel={close}
+                    />
+                );
+        }
+    };
+
```

Comment:
Unless we expect more steps than this in the future, I think this is small enough and nearly-only-components enough that we could move this rendering code right into the <Modal /> rendering below with just `step === ParticipationStep.ConfirmParticipation && ... <ConfirmParticipationStep ... />` and `step === ...ConfirmPassword ... <ConfirmPasswordStep ... />`

optional, but I think it reads better together